modules/apps/adaptive-media/adaptive-media-image-api/src/main/java/com/liferay/adaptive/media/image/counter/BaseAMImageCounter.java

```diff
@@ -50,46 +61,107 @@ public int countExpectedAMImageEntries(long companyId) {
 		return counter.get();
 	}
 
-	protected abstract void forEachFileEntry(
-			long companyId, Consumer<DLFileEntry> consumer)
-		throws PortalException;
-
-	protected String[] getMimeTypes() {
-		return ArrayUtil.filter(
-			amImageMimeTypeProvider.getSupportedMimeTypes(),
-			amImageValidator::isProcessingSupported);
-	}
+	protected abstract long getClassNameId();
 
 	@Reference
 	protected AMImageMimeTypeProvider amImageMimeTypeProvider;
 
 	@Reference
 	protected AMImageValidator amImageValidator;
 
+	@Reference
+	protected ClassNameLocalService classNameLocalService;
+
 	@Reference
 	protected DLFileEntryConfigurationProvider dlFileEntryConfigurationProvider;
 
-	private Consumer<DLFileEntry> _getCountDLFileEntryConsumer(
-		AtomicInteger counter) {
+	@Reference
+	protected DLFileEntryLocalService dlFileEntryLocalService;
+
+	private void _forEachGroup(long companyId, Consumer<Group> groupConsumer)
+		throws PortalException {
+
+		ActionableDynamicQuery actionableDynamicQuery =
+			GroupLocalServiceUtil.getActionableDynamicQuery();
+
+		actionableDynamicQuery.setModelClass(Group.class);
+
+		actionableDynamicQuery.setCompanyId(companyId);
+
+		actionableDynamicQuery.setPerformActionMethod(
+			(ActionableDynamicQuery.PerformActionMethod<Group>)
+				groupConsumer::accept);
+
+		actionableDynamicQuery.performActions();
+	}
+
+	private Consumer<Group> _getCountDLFileEntryConsumer(
+		long companyId, AtomicInteger counter,
+		long previewableProcessorMaxSize) {
 
 		Map<Long, Long> groupPreviewableProcessorMaxSizeMap =
 			dlFileEntryConfigurationProvider.
 				getGroupPreviewableProcessorMaxSizeMap();
 
-		return dlFileEntry -> {
-			Long previewableProcessorMaxSize =
-				groupPreviewableProcessorMaxSizeMap.get(
-					dlFileEntry.getGroupId());
+		return group -> {
+			long classNameId = getClassNameId();
+
+			long groupId = group.getGroupId();
+
+			Long previewableGroupProcessorMaxSize =
+				groupPreviewableProcessorMaxSizeMap.getOrDefault(
+					groupId, previewableProcessorMaxSize);
+
+			DSLQuery dslQuery = DSLQueryFactoryUtil.count(
+			).from(
+				DLFileEntryTable.INSTANCE
+			).leftJoinOn(
+				TrashEntryTable.INSTANCE,
+				TrashEntryTable.INSTANCE.classPK.eq(
+					DLFileEntryTable.INSTANCE.fileEntryId)
+			).where(
+				DLFileEntryTable.INSTANCE.groupId.eq(
+					groupId
+				).and(
+					() -> {
+						if (classNameId > 0) {
+							return DLFileEntryTable.INSTANCE.classNameId.eq(
+								getClassNameId());
+						}
+
+						return DLFileEntryTable.INSTANCE.repositoryId.eq(
+							groupId);
```

Comment:
The class name id is only used here, so it is not necessary to read it at the top of the method. Change it to something like:
```
if (Validator.isNotNull(getClassName()) {
    return DLFileEntryTable.INSTANCE.classNameId.eq(classNameLocalService.getClassNameId(getClassName()));
}

...
```
This way, we only invoke the service if it is strictly necessary.